djangocms_pageadmin/admin.py

```diff
@@ -53,15 +55,36 @@ def get_list_display(self, request):
     def get_queryset(self, request):
         """Filter PageContent objects by current site of the request.
         """
+        url_subquery = PageUrl.objects.filter(
+            language=OuterRef("language"), page=OuterRef("page")
+        )
+        draft_version_lock_subquery = VersionLock.objects.filter(
```

Comment:
I think we should describe the reason that this is here to help anyone reading this in the future. "Collect locked status to handle  the requirement that lock on a draft version dictates the unpublish permission on a published version".